mozetl/search/aggregates.py

```diff
@@ -192,16 +194,28 @@ def add_derived_columns(exploded_search_counts):
     '''
     udf_get_search_addon_version = udf(get_search_addon_version, StringType())
 
-    return (
-        exploded_search_counts
-        .withColumn(
-            'type',
-            when(col('source').startswith('sap:'), 'tagged-sap')
+    def _generate_when_expr(sources):
```

Comment:
I think this can be simplified with pyspark's [`isin`](http://spark.apache.org/docs/2.1.0/api/python/pyspark.sql.html#pyspark.sql.Column.isin) function.

Reply:
yup, I didn't think that would work but misunderstood how sap sources were defined.